apps/aecore/src/aec_peer_connection.erl

```diff
@@ -378,6 +370,35 @@ handle_msg(S, MsgType, true, Request, {ok, {MsgType, _Vsn, Msg}}) ->
         block   -> handle_get_block_rsp(S, Request, Msg)
     end.
 
+handle_request(S, {Request, Args}, From) ->
+    case get_request(S, Request) of
+        none ->
+            handle_request(S, Request, Args, From);
+        {Request, _From} ->
+            {reply, {error, request_already_in_progress}, S}
```

Comment:
Are all requests in this state guaranteed to eventually be removed e.g. in case of process crash or slow other TCP endpoint?